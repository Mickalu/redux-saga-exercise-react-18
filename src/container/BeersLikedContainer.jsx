import React from 'react';
import { useSelector } from 'react-redux';

import { findBeerById } from '../utils/findBeerById';

export const BeersLikedContainer = () => {
  const beerslikedState = useSelector((state) => state.beersLiked);
  const beersState = useSelector((state) => state.beers);

  return (
    <>
      {beerslikedState.data.map((beerId) => {
        const beerTitle = findBeerById(beerId, beersState.data).title;
        return (
          <div data-testid="beer-title" key={`div-beer-liked-${beerId}`}> {beerTitle} </div>
        );
      })}
    </>
  );
};

export default BeersLikedContainer;
