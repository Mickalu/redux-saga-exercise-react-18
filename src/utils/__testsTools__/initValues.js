export const initListBeers = {
  data:
    [
      {
        id: "82",
        title: "Karmeliet Triple",
        description: "<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>",
        country: "Belgium",
        type_beer: "Tripel",
        company: "Brasserie Bosteels",
        colour: "Blond",
        graduation: "6 %",
        size: "33 cl",
        price: "2.5",
        photo_link: "https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg",
      },
      {
        id: "126",
        title: "Cuvée des trolls",
        description: "<p>A tradionally made beer with fruity notes of orange and honey. Discover one of the reference beers from Belgium!</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Dubuisson",
        colour: "Blond",
        graduation: "8 %",
        size: "25 cl",
        price: "1.6",
        photo_link: "https://img.saveur-biere.com/img/p/126-14990-thickbox_sb3.jpg",
      },
      {
        id: "104",
        title: "Kwak",
        description: "<p>This Belgian beer no longer needs any introduction. Best poured in the original coachman's glass from this brand!</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Bosteels",
        colour: "Amber",
        graduation: "8 %",
        size: "33 cl",
        price: "2.1",
        photo_link: "https://img.saveur-biere.com/img/p/104-16465-thickbox_sb3.jpg",
      },
      {
        id: "81",
        title: "Orval",
        description: "<p>Orval is a triple trappist bitter-sweet beer with an evolutive flavour! A true Belgian reference beer.</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Abbaye d'Orval",
        colour: "Amber",
        graduation: "6 %",
        size: "33 cl",
        price: "2.8",
        photo_link: "https://img.saveur-biere.com/img/p/81-14420-thickbox_sb3.jpg",
      },
      {
        id: "91",
        title: "Rochefort 10",
        description: "<p>A Belgian Trappist beer with a plum and cacao taste! A Marvellous product with a 11.3° ABV. </p>",
        country: "Belgium",
        type_beer: "Quadrupel",
        company: "Abbaye de Rochefort",
        colour: "Dark",
        graduation: "10 %",
        size: "33 cl",
        price: "3.4",
        photo_link: "https://img.saveur-biere.com/img/p/91-14652-thickbox_sb3.jpg",
      },
      {
        id: "101",
        title: "Lindemans Faro",
        description: "<p>Discover a Belgian fruit beer, extremely refreshing and sweet. Maybe the most popular beer among women!</p>",
        country: "Belgium",
        type_beer: "Faro",
        company: "Brasserie Lindemans",
        colour: "Amber",
        graduation: "8 %",
        size: "25 cl",
        price: "1.8",
        photo_link: "https://img.saveur-biere.com/img/p/101-14699-thickbox_sb3.jpg",
      },
      {
        id: "100",
        title: "Lindemans Pêcheresse",
        description: "<p>Discover this beer with peach flavors to be poured in a flute-shaped glass. A nice cocktail for the ladies!</p>",
        country: "Belgium",
        type_beer: "Lambic - Fruit",
        company: "Brasserie Lindemans",
        colour: "Blond",
        graduation: "9 %",
        size: "25 cl",
        price: "2.2",
        photo_link: "https://img.saveur-biere.com/img/p/100-14012-thickbox_sb3.jpg",
      },
      {
        id: "736",
        title: "Lindemans Kriek",
        description: "<p>Discover a beer flavored with cherries, highly popular among women!</p>",
        country: "Belgium",
        type_beer: "Lambic - Fruit",
        company: "Brasserie Lindemans",
        colour: "Ruby",
        graduation: "4 %",
        size: "25 cl",
        price: "1.9",
        photo_link: "https://img.saveur-biere.com/img/p/736-14700-thickbox_sb3.jpg",
      },
      {
        id: "99",
        title: "Lindemans Framboise",
        description: "<p>Discover a beer with a strong taste of raspberry and a fine balance between acidity and fruity sweetness!</p>",
        country: "Belgium",
        type_beer: "Lambic - Fruit",
        company: "Brasserie Lindemans",
        colour: "Ruby",
        graduation: "6 %",
        size: "25 cl",
        price: "2.1",
        photo_link: "https://img.saveur-biere.com/img/p/99-14010-thickbox_sb3.jpg",
      },
      {
        id: "58",
        title: "Kasteel Bruin",
        description: "<p>Discover a strong Belgian beer with roasted notes, hints of moka, liquorice and banana!</p>\n<p> </p>",
        country: "Belgium",
        type_beer: "Belgian dark ale / Brune",
        company: "Brasserie Van Honsenbrouck",
        colour: "Dark",
        graduation: "9 %",
        size: "33 cl",
        price: "2.6",
        photo_link: "https://img.saveur-biere.com/img/p/58-16778-thickbox_sb3.jpg",
      },
      {
        id: "90",
        title: "Rochefort 8",
        description: "<p>An exceptional Belgian Trappist beer with, jammy stone-fruit and spice flavours</p>",
        country: "Belgium",
        type_beer: "Belgian dark ale / Brune",
        company: "Abbaye de Rochefort",
        colour: "Dark",
        graduation: "9 %",
        size: "33 cl",
        price: "2.7",
        photo_link: "https://img.saveur-biere.com/img/p/90-16527-thickbox_sb3.jpg",
      },
      {
        id: "103",
        title: "Honey Barbar",
        description: "<p>Taste all the flavours and the charm of sweet honey in this Honey Barbar.</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Lefebvre",
        colour: "Blond",
        graduation: "7 %",
        size: "33 cl",
        price: "2.4",
        photo_link: "https://img.saveur-biere.com/img/p/103-16445-thickbox_sb3.jpg",
      },
      {
        id: "110",
        title: "Delirium Tremens",
        description: "<p>The most successful product of the brewery Huyghe in a 33cl bottle !</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Huyghe",
        colour: "Blond",
        graduation: "6 %",
        size: "33 cl",
        price: "3.2",
        photo_link: "https://img.saveur-biere.com/img/p/110-16755-thickbox_sb3.jpg",
      },
    ],
};

export const initBeersState = {
  beers: {
    data: [
      {id:"82", title:"Karmeliet Triple", description:"<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>", country:"Belgium", type_beer:"Tripel", company:"Brasserie Bosteels", colour:"Blond", graduation:"6 %", size:"33 cl", price:"2.5", photo_link:"https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg"},
      {id:"126", title:"Cuvée des trolls", description:"<p>A tradionally made beer with fruity notes of orange and honey. Discover one of the reference beers from Belgium!</p>", country:"Belgium", type_beer:"Belgian Pale ale / Blonde", company:"Brasserie Dubuisson", colour:"Blond", graduation:"8 %", size:"25 cl", price:"1.6", photo_link:"https://img.saveur-biere.com/img/p/126-14990-thickbox_sb3.jpg"}
    ],
    isFetching: false,
  },
};

export const initCurrentIndexState = {
  currentIndex: 0,
};

export const initCurrentIndexLastindex = (listTest) => (
  { currentIndex: listTest.length - 1}
);

export const initBeerState = {
  id:"82", title:"Karmeliet Triple", description:"<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>", country:"Belgium", type_beer:"Tripel", company:"Brasserie Bosteels", colour:"Blond", graduation:"6 %", size:"33 cl", price:"2.5", photo_link:"https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg"
};

export const initBeersSliceState = {
  data: [],
  isFetching: false,
};
