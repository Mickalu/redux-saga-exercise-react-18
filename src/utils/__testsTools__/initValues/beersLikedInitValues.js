export const INIT_BEERS_LIKED_STATE_VALUE = {
  data: [],
};

export const beersLikedStateWithIds = (listIds) => ({data: listIds});
