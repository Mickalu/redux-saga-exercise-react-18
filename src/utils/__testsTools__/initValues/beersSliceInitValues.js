export const INIT_BEERS_SLICE_STATE = {
  data: [],
  isFetching: false,
};

export const initBeersSliceWithBeers = {
  data: [
    {id:"82", title:"Karmeliet Triple", description:"<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>", country:"Belgium", type_beer:"Tripel", company:"Brasserie Bosteels", colour:"Blond", graduation:"6 %", size:"33 cl", price:"2.5", photo_link:"https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg"},
    {id:"126", title:"Cuv√©e des trolls", description:"<p>A tradionally made beer with fruity notes of orange and honey. Discover one of the reference beers from Belgium!</p>", country:"Belgium", type_beer:"Belgian Pale ale / Blonde", company:"Brasserie Dubuisson", colour:"Blond", graduation:"8 %", size:"25 cl", price:"1.6", photo_link:"https://img.saveur-biere.com/img/p/126-14990-thickbox_sb3.jpg"}
  ],
  isFetching: false,
}

export const stateIsFetchingTrue = {
  data: [],
  isFetching: true,
};
