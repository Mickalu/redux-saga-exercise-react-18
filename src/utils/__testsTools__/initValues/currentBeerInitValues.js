export const INIT_CURRENT_BEER_STATE = {
  id: null,
};
