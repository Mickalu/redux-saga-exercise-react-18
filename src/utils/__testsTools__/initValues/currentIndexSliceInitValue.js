export const INIT_CURRENT_INDEX_STATE = {
  currentIndex: 0,
};

export const currentIndexIncrementedByOne = {
  currentIndex: 1,
};
