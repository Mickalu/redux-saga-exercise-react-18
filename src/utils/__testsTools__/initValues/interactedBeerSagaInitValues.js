export const paraminteractionLikeBeerTrue = {
  payload: {
    isLiked: true,
  },
};

export const paramInteractedBeerSagaFalse = {
  payload: {
    isLiked: false,
  },
};
