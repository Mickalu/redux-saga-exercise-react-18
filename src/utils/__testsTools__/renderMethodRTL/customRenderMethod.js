import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";

import { store } from "../../../store";

const getByTextCustom = (text) => (
  screen.getByText(text)
);

const queryAllByRoleCustom = (role, options) => (
  screen.queryAllByRole(role, options)
);

const queryByTestIdCustom = (testId) => (
  screen.queryByTestId(testId)
);

const getByTestIdCustom = (testId) => (
  screen.getByTestId(testId)
);

const getByDisplayValueCustom = (displayValue) => (
  screen.getByDisplayValue(displayValue)
);

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    { children }
  </Provider>
);

export const renderCustom = (ui, options = {}) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

export { renderCustom as render };
export { getByTextCustom as getByText };
export { queryAllByRoleCustom as queryAllByRole };
export { queryByTestIdCustom as queryBytestId };
export { getByTestIdCustom as getByTestId };
export { getByDisplayValueCustom as getByDisplayValue };
