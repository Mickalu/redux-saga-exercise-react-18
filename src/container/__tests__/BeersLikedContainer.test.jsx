import React from "react";
import { useDispatch, useSelector } from 'react-redux';

import { render, queryBytestId } from "../../utils/__testsTools__/renderMethodRTL/customRenderMethod";
import { initListBeers } from "../../utils/__testsTools__/initValues";
import BeersLikedContainer from "../BeersLikedContainer";


const reactRedux = { useDispatch, useSelector };

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn()
}));

it("Should not have beer title at the beginning", () => {
  useSelector.mockImplementation(callback => {
    return callback({
      beersLiked: { data: [] },
      beers: initListBeers,
    });
  });

  render(<BeersLikedContainer />);

  expect(queryBytestId("beer-title")).not.toBeInTheDocument();
});

it("beersLikedContainer Should match snapshot", () => {
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");

  useDispatchMock.mockReturnValue(jest.fn());
  useSelectorMock.mockImplementation(callback => {
    return callback({
      beersLiked: { data: ["126"] },
      beers: initListBeers,
    });
  });

  const { container } = render(<BeersLikedContainer />);

  expect(container).toMatchSnapshot();
});

it("beersLikedContainer with multiple liked beer should display all title", () => {
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");

  useDispatchMock.mockReturnValue(jest.fn());
  useSelectorMock.mockImplementation(callback => {
    return callback({
      beersLiked: { data: ["126", "82"] },
      beers: initListBeers,
    });
  });

  const { container } = render(<BeersLikedContainer />);
  expect(container).toMatchSnapshot();
});
