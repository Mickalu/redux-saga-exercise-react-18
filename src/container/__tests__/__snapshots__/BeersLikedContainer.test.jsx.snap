// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`beersLikedContainer Should match snapshot 1`] = `
<div>
  <div
    data-testid="beer-title"
  >
     
    Cuvée des trolls
     
  </div>
</div>
`;

exports[`beersLikedContainer with multiple liked beer should display all title 1`] = `
<div>
  <div
    data-testid="beer-title"
  >
     
    Cuvée des trolls
     
  </div>
  <div
    data-testid="beer-title"
  >
     
    Karmeliet Triple
     
  </div>
</div>
`;
